src/Symfony/Component/Validator/Tests/Constraints/TimeValidatorTest.php

```diff
@@ -62,6 +69,50 @@ public function getValidTimes()
         ];
     }
 
+    /**
+     * @dataProvider getValidTimesWithoutSeconds
+     */
+    public function testValidTimesWithoutSeconds($time)
+    {
+        $this->validator->validate($time, new Time([
+            'withSeconds' => false,
+        ]));
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidTimesWithoutSeconds()
+    {
+        return [
+            ['01:02'],
+            ['00:00'],
+            ['23:59'],
+        ];
+    }
```

Comment:
I would suggest using `yield` and generators on your data providers, to keep consistency with other tests 👍 

```suggestion
    public function getValidTimesWithoutSeconds(): \Generator
    {
        yield ['01:02'];
        yield ['00:00'];
        yield ['23:59'];
    }
```

Reply:
What consistency are you talking about? There are no generators here.

Reply:
Consistency with other tests (maybe not in this class, that's right). I don't know if a maintainer knows if there's a rule about data providers and generators in the Symfony codebase?

But one thing for sure is that non-static data providers will be deprecated in PHPUnit 10. This could be a good idea to declare it static in any case 🙂